cob_twist_controller/src/constraint_solvers/solvers/task_priority_solver.cpp

```diff
@@ -64,8 +64,9 @@ Eigen::MatrixXd TaskPrioritySolver::solve(const Vector6d_t& in_cart_velocities,
         predict_jnts_vel.q(i) = particular_solution(i, 0) * cycle + joint_states.current_q_(i);
         predict_jnts_vel.qdot(i) = particular_solution(i, 0);
     }
+    Eigen::FullPivLU<Eigen::MatrixXd> lu_decomp(projector);
 
-    if (this->constraints_.size() > 0)
+    if ((this->constraints_.size() > 0) && (lu_decomp.rank() != 0))
```

Comment:
Can you put the check into a new ```if```?
Like this:
```
if (this->constraints_.size() > 0)
{
    if(lu_decomp.rank() != 0)
    {
```
because than it's more clear was has happened, i.e. whether the first or the second condition is ```false```